### PR TITLE
Bind vips_max with default size to VipsImage.max1 + class for the return values

### PR DESCRIPTION
**Pull request by stuart12**
Bind to vips_max (the simple version with size set to 1) and test.

### CODE REVIEW

File: src/main/c/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-file(GLOB SOURCES Vips.c VipsContext.c VipsException.c VipsImage.c)
+file(GLOB SOURCES Vips.c VipsContext.c VipsException.c VipsImage.c Max1Result.c)
 
 add_library(JVips SHARED ${SOURCES})
 add_dependencies(JVips Vips_header VipsContext_header VipsImage_header)
```
Comment (dbouron) [inline src/main/c/CMakeLists.txt]: You have to add a dependency on Max1Result_header too

File: src/main/c/fields.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+extern jfieldID field_Max1Result_out;
+extern jfieldID field_Max1Result_x;
+extern jfieldID field_Max1Result_y;
```
Comment (dbouron) [inline src/main/c/fields.h]: I think you can put it directly in VipsImage.c and avoid to include field.h

File: src/main/java/com/criteo/vips/Max1Result.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  limitations under the License.
+*/
+
+package com.criteo.vips;
+
+public class Max1Result {
+    int x;
+    int y;
+    double out;
+    private static native void initMax1ResultFieldIDs();
```
Comment (dbouron) [inline src/main/java/com/criteo/vips/Max1Result.java]: OPT: move after the static block

File: src/main/java/com/criteo/vips/VipsImageImpl.java [hunk 4]
```diff
@@ -147,6 +147,14 @@ public void resize(Dimension dimension, boolean scale) throws VipsException {
 
     private native void resizeNative(int width, int height, boolean scale) throws VipsException;
 
+    public Max1Result max1() throws VipsException {
+        Max1Result r = new Max1Result();
+        max1Native(r);
+        return r;
+    }
```
Comment (dbouron) [inline src/main/java/com/criteo/vips/VipsImageImpl.java]: OPT: you can create the Max1Result object in the native method. Thus, you don't need this method but only `max1Native(Max1Result r)`

File: src/main/c/fields.h (hunk 2 above)
Comment (stuart12) [reply src/main/c/fields.h]: yes but then the compiler doesn't check that the declarations match

File: src/main/c/CMakeLists.txt (hunk 1 above)
Comment (stuart12) [reply src/main/c/CMakeLists.txt]: done

File: src/main/java/com/criteo/vips/Max1Result.java (hunk 3 above)
Comment (stuart12) [reply src/main/java/com/criteo/vips/Max1Result.java]: done

File: src/main/java/com/criteo/vips/VipsImageImpl.java (hunk 4 above)
Comment (stuart12) [reply src/main/java/com/criteo/vips/VipsImageImpl.java]: I wanted to do as little as possible in the native method. It seems cleaner to create a Java object in Java code.

**Pull request merged**